snuba/subscriptions/data.py

```diff
@@ -161,18 +166,21 @@ def build_request(
             dataset,
             timer,
             referrer,
-            [
-                self.entity_subscription.validate_query,
-                partial(self.add_conditions, timestamp, offset),
-            ],
+            custom_processing,
         )
         return request
 
     @classmethod
     def from_dict(
         cls, data: Mapping[str, Any], entity_key: EntityKey
     ) -> SubscriptionData:
-        entity_subscription = get_entity_subscription(entity_key)(data_dict=data)
+        entity_subscription = get_entity(entity_key).get_entity_subscription()
+        assert entity_subscription is not None
+
+        metadata = {}
+        for key in data.keys():
+            if key not in {"project_id", "time_window", "resolution", "query"}:
```

Comment:
Can you define these hardcoded keys as a constant with an explanation of why they are not part of metadata?